### ISSUE TITLE
gd does not support transparency 

### ISSUE THREAD

**Issue by rosell-dk** [rosell-dk/webp-convert#42]
It seems our gd converter does not support the alpha channel when converting png's. Transparent pixels come out white.

We should investigate if it is possible with gd.
And update the README

**Comment by rosell-dk** [comment]
btw, it does not seem to have any issues with large filesizes, as claimed in the README.

**Comment by rosell-dk** [comment]
Perhaps the stuff going on in this class can help us on the way?
https://github.com/PHPixie/Image/blob/master/src/PHPixie/Image/Drivers/Type/GD.php

**Comment by rosell-dk** [comment]
https://stackoverflow.com/questions/1705098/imagepng-and-transparency-in-gd-library-with-php

**Comment by rosell-dk** [comment]
It seems the "media webp" Wordpress plugin manages to handle transparency with Gd.

In `classes/class-admin.php`:
```
if ( 'image/png' === $image_details['mime'] ) {
    self::$_img = imagecreatefrompng( $filearray['file'] );
} elseif ( 'image/jpeg' === $image_details['mime'] ) {
    self::$_img = imagecreatefromjpeg( $filearray['file'] );
}

imagepalettetotruecolor( self::$_img );
imagealphablending( self::$_img, true );
imagesavealpha( self::$_img, true );

if ( ! $ignore ) {
    imagewebp( self::$_img, $filearray['file'] . '.webp' );
}
```

**Comment by rosell-dk** [comment]
When testing this out, I discovered that imagewebp() only handles true color images. The `imagepalettetotruecolor` call fixes that :)

**Comment by rosell-dk** [comment]
Fixed both in master (the upcoming 2.0) and created a new 1.3.4 release for the 1.3 release series

**Comment by LeoRichard** [comment]
Hello, i got problems with PNGs in Safari. Im using Cache enabler (already checked the webp cache option). Also on edge when opening the image itself it shows a weird code, not the image:
`��WEBPVP8LI
��/ü�abÿuâ]Ïó\wwwwwwwwwwwwwwww÷{ÿvÝÝÝýþäœéÿy˜ëË!Íì
„Ý„Iw“†.¡¤;`ÒÕçrVÍÌ®ùMÙ‘†Y2iØv„9ì&–„“ÙæÃné’æŸÙMJf7¥a×4‡DœÌlþÕñý¿þÿ÷ÿõ?ü;ëïÿ³ÿðxìOþð/ø¡ßí;~›ÑÛ}çð3çÞã¬_øÙ{¬µ‘ÜZû137&rwþç.ýì Cõë?md£·¬Y;ôcÀÕ<îNíA?Ó ãäÓ¸5.Z[K³ð‰ßëót'Âœî0;62îÚþ_z¬˜)M¸SéùÒì(W>¼5®ÛÒ¡ƒ¿œ
YüÁ•Õ/¾òíÉ Zýé?¶š	ÓÕ›>™ÎŒúœ¨Ns´š¹9Zý¾åQ0dCûæ5LTšÐûžxøe¾r˜g
àusÒ|ïï*ùµñ(¾`”ø8!ýø¡ð[ã°™íiªŒ}I[h?Åþ—
¬Ýö©O<TÌLìU½Ø‰ÇŒöæælGhÙ§oòg³htíGý‡z3Âb’þÓF“aøraüÈº?ÜÁz6d¸h#g‡äJSÃØ5^+É¼F’/–tFÖ°žcØ?k+ãžNTJ:óH¾~E$yÂN&Iz÷¥'æ:ØI$‰ÔÝê(‘H$>hÌZ£lzr…ç5ÆÎÆÂb–ü¼±Ò[�pÙ‰�îV�8s
Ï,}ÊÚäädmÇJìäÎJÜ“E¥ô1»‚üƒg+å�OÖ‰êw«ëB"Ö~½5bÓÔ§}Js=ÂÆû&Yù–%O0¶OgªÌWˆ7Jòœ!™€!Œ`æåŽLÝßGðù#.é}}I3~[ªWKdrÈ“œUUgF’ž±$ÍŒGr`æ"In�×êsO6à©™y’õ™½ŽÐYItDrõè¶c‡¾ªgtâ“Ìö71Q¼¢+2HfIf“Y’A2+0ØNå˜¬Ar•d.™fÇ'(EÆ¯›ä•ž¦çÙ¬bÜó<[_%µím÷;ÅÈc.æÈ¯ºoŒ1%cÌþÙYó„µã’QkŒñÈbÓ¸aö×ÈÛJêÇ%)ÉurÖs§Ùï3m…sZáÄöw.Æ^rôÊöÄªk¾†?Ñ·}:ñOV¯TOúŽ‚Ûöõ¥c ëë;
\êšä¸K&0×#liTæ7
@á¹¶9ÑjµžÿÉîJ«U™”mTZ­J†äÅJ«Õ:”ù_ñ%ÏåÂz4æä_}þ­V«R•°þ¬
@áÚRZ§¼´7à!>Ãþ²Ö•«‹y ¿xµÛÛžhµZ­Õ"ÉÌÀbxàƒ»Ûîy‡ä9·JY®:é¾
e”�Kr�â²a�X&Ù€”¤ø"¡žê_ÕÊ€h}ÙÖ=P®§TõE¨/¨ÊW¡yª×QrâÉ¥”y/p¡*Ù&»]ò6Èr$ì(–
ÐÜË(‘¬	=2Oøú$_²P/C·‘ÑjB&%.hvõÉ†Ð’ýúkÐÞ<é ×‚¸Dò´']¸­d‡
zšîŒ‘)O8)ÄÆÜ£o@
ÁS‡´ë!ÈORgâjN!6Rs,–……Ä©Å.IcVÈ=Wˆ¹T^À³þëzû#’éE�Kï½t£D¡
¿ÓFÆ[&vÜkjmxÆxÞé5…ï,Û…ør¿ïN§^rìœ†øü»{‹®5v¨³Ð¦ÓÔy¹ÕôÞ^`H¦/ÀÜÖ¯¦w=�,VHž…x*“NgÎÒ‚xÆ'y�zH²s$¢›¹Ó‚wŽd²ÞÑÆU“¥‹§Ï’ìóŒ1&:’
Â>E V*sÂ‰I¦»©ÁÑ.�XŸ¦Î>E@˜I’[�0sâÍG
�dvD¦´)
yáùgI²&ÔþaßîXcŒñ–IÒÏÍFª˜Ë…$yÛÈc¢‹«’-�
$LÆÈW€©#êJžzq�ænRg$°’€²%ôS>$Ü"¿3�¤*²`�ºWoNÀD’â²ZùÙ!]¯JŒw! ÉÙ•~_dD;Ié°QÙ#%a•Î.?k�˜é£Ö<•žÐÃ  p¤8ùÌU8$ìSi„yÅùu�9¢´<±ðÔ½îÐ¥ 	ÆëiÓÙºìzV
~}ßgd7e—„
ªåÙN» 7ÔëVX|ˆ�Fþº"· pÄ¦°¥Úöò-*W$âÞš;¬[‰1^3^²%ÁÄKRã54êÏÉZÂ‚ª/FN	K.¤…®ï«WUÕ„3<*�x²¡"Ü~6=aRuAVŒÌ—§ÜZT ¿ãNo<’IeÎí6å§„UÍA\6ìlXhºpmíÀHYkV5/4YîðH+Šô€©i–„Óª%Á*ÎÔ<ÿœ‚¸vjP‚‡xè
ì±³K9EI¸¥:å`XvÃÙŠ°:c¿€„¯³«º!Œ3ÜÐèTt7�Œ…¼ ÜPÝ&	î
ˆ«HfSÍ�Ì»œ¶:fÚÈNPÙ)ä¿ªì;?pÕpMRm;Ë4�4j.„á%ó:…”¬Þ�%=�¸Jü�ì“Å€ü²l- p¤Q³²¥KÂ=îð¶6Òð¶Â°Ç‘ýªTƒ�ðH;…ï¼Õh@£F²øÔá,Ü€™5!µ›Ôáô¢ïÔÀ`JhwÀ
ŸÌä`? ™m@>E2�s}Âö£€Ujðû�`æ¶BùÆŠdRðHú¾3vÛHešCËq'v+ÔàŠ€ü¥þ¥Ä‡Æ_€ÂÀŠ÷Há—�O,ÍŸhãùgu¸œ€G_Ô@áå.õ·�Ðè$É3Ö£;ñ*IŽ� ½º´”(�@!£Åç2ù¬ÖÁÉòWvÈë;‡ÎØ¶‘Ê4=ãÐžÍR÷¯ßôÎ
ò¹†3A:,	x®]–àÐõ(þÎuAs½,pEÐ¢žŸ0À`âTÄ¹2““§ÏÖgì´‘Êid·²Ôÿõ×4æv5‚óª—<BO
kZ¹ç*ÀÈ¦V¢ÆnNÂŸ}Më©ÿlÊûó:ùy:àï¼,`+ÈóuòèëGãk½.ðû–¬;QtÒir¸ »uÛY
N·dg²5a™ä%!%õMÕSÿªdïCpYàÏîpŽM�(vÈýYŸÊtÔ¥˜Šþ0Õk{ª½5J3Â	Ýy!…6Uë
$ÃÉúdÝwƒÓãÖ:‹lÿ(Ý¼9ŸØ{®û_ßgM‡~{¸¼·ÛIN/´Û
¿“du¡Ýn§eüëöÄÞÞKŽ¯åHæ:ÛíöJo.´Ûí…nŽ-´Ûí,+çNíÝØÝ)S»¸qõüæù«Ej‡ÝÑóßÜ¬Dk!åév»½‘p{¡ÝnTÉ¿ÞµžúÄ­áÙ€b.s=¤»~fÈÚH'²öbg–ÇVï¾~HWÃ1éûŒÉp¬v1²êøø÷
x¬ï§ÜoÌþìîúÖÆÆxmùæf
wÄúÿÌ›<¾«d®_Ïô·ý¿§�`
 Im stuck on this, already cleared cache multiple times. Any ideas?
